File: Classes/HomeViewController.m
//
//  HomeViewController.m
//  climbingweather
//
//  Created by Jonathan StJohn on 1/16/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "HomeViewController.h"


@implementation HomeViewController

- (id) init
{
	// Call the super-class's designated initialize
	[super initWithNibName: @"HomeViewController" bundle: nil];
	
	// Get tab bar item
	UITabBarItem *tbi = [self tabBarItem];
	
	// Give it a label
	[tbi setTitle: @"Home"];
	
	// Add image
	UIImage *i = [UIImage imageNamed:@"icon_home.png"];
	
	// Put image on tab
	[tbi setImage: i];
	
	[[self navigationItem] setTitle: @"Home"];
	
	return self;
}

// Did appears
- (void) viewDidAppear: (BOOL) animated
{
	[[self tabBarController] setTitle: @"Home"];
	[[self navigationController] setNavigationBarHidden: YES];
}

// The designated initializer.  Override if you create the controller programmatically and want to perform customization that is not appropriate for viewDidLoad.
/*
- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization.
    }
    return self;
}
*/

/*
// Implement viewDidLoad to do additional setup after loading the view, typically from a nib.
- (void)viewDidLoad {
    [super viewDidLoad];
}
*/

/*
// Override to allow orientations other than the default portrait orientation.
- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation {
    // Return YES for supported orientations.
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}
*/

- (void)didReceiveMemoryWarning {
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc. that aren't in use.
}

- (void)viewDidUnload {
    [super viewDidUnload];
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

- (void) searchBarSearchButtonClicked:(UISearchBar *)searchBar
{
	[searchBar resignFirstResponder];
	NSString *searchBarText = [NSString stringWithFormat: @"%@", [searchBar text]];
	[[[[self tabBarController] viewControllers] objectAtIndex: 4] setInitialSearch: searchBarText];
	[searchBarText release];
	[[self tabBarController] setSelectedIndex: 4];
}

- (IBAction) showNearby: (id) sender
{
	[[self tabBarController] setSelectedIndex: 1];
}

- (IBAction) showByState: (id) sender
{
	[[self tabBarController] setSelectedIndex: 2];
}

- (IBAction) showFavorites: (id) sender
{
	[[self tabBarController] setSelectedIndex: 3];
}

- (IBAction) showSearch: (id) sender
{
	[[self tabBarController] setSelectedIndex: 4];
}

- (void)dealloc {
    [super dealloc];
}


@end

File: HomeViewController.swift
//
//  HomeViewController.swift
//  climbingweather
//
//  Created by Jon St. John on 2/22/17.
//
//

import Foundation

class HomeViewController: UIViewController, UITextFieldDelegate {
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
    }
    
    override func viewWillAppear(_ animated: Bool) {
        self.navigationController?.isNavigationBarHidden = true
    }
    
    @IBAction func showNearby(sender: UIButton) {
        self.tabBarController?.selectedIndex = 1
    }
    
    @IBAction func showByState(sender: UIButton) {
        self.tabBarController?.selectedIndex = 2
    }
    
    @IBAction func showFavorites(sender: UIButton) {
        self.tabBarController?.selectedIndex = 3
    }
    
    @IBAction func showSettings(sender: UIButton) {
        
        let tabBarController = UITabBarController()
        
        let settingsViewController = SettingsViewController()
        let settingsTabImage = UIImage(named: "20-gear2.png")
        settingsViewController.tabBarItem = UITabBarItem(title: "Settings", image: settingsTabImage, selectedImage: settingsTabImage)
        tabBarController.viewControllers = [AboutViewController(), settingsViewController]
        self.navigationController?.pushViewController(tabBarController, animated: false)
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {   //delegate method
        textField.resignFirstResponder()
        
        if let searchController = self.tabBarController?.viewControllers?[4] as? AreasViewController {
            searchController.search = .Term(textField.text ?? "")
        }
        
        self.tabBarController?.selectedIndex = 4
        
        return true
    }
    
}
